app/utils/auth-storage/auth-storage.server.ts

```diff
@@ -0,0 +1,47 @@
+import { createCookieSessionStorage } from "@remix-run/node";
+import invariant from "tiny-invariant";
+import type { PersistSessionParams } from './types'
+import { createSession } from '~/models/session'
+
+invariant(process.env.SESSION_SECRET, "SESSION_SECRET must be set");
+
+const ACCESS_TOKEN_KEY = "accessToken";
+const EXPIRATION_TIME = 60 * 60 * 24 * 7 // 1 week
+const sessionStorage = createCookieSessionStorage({
+  cookie: {
+    name: "__auth_session",
+    httpOnly: true,
+    path: "/",
+    sameSite: "lax",
+    secrets: [process.env.SESSION_SECRET],
+    secure: process.env.NODE_ENV === "production",
+  },
+});
```

Comment:
this module will handle saving the auth access token in a browser cookie. The cookie will be HttpOnly, and will be constrained to the app domain (sameSite policy). So, the access token won't be accessible from any js script on the browser, only the frontend server will be able to read it because the browser will send it on each request